gsy_framework/sim_results/electric_blue/all_results.py

```diff
@@ -0,0 +1,103 @@
+from collections import defaultdict
+from typing import Dict
+
+from pendulum import from_format, DateTime, Duration
+
+from gsy_framework.constants_limits import DATE_TIME_FORMAT
+from gsy_framework.enums import AvailableMarketTypes
+from gsy_framework.forward_markets.forward_profile import ALLOWED_MARKET_TYPES
+from gsy_framework.sim_results.electric_blue.aggregate_results import (handle_forward_results,
+                                                                       ForwardDeviceStats,
+                                                                       MARKET_RESOLUTIONS)
+from gsy_framework.sim_results.electric_blue.time_series import ForwardDeviceTimeSeries
+
+
+class ForwardResultsHandler:
+    """Calculate all results for each market slot for forward markets."""
+
+    def __init__(self, slot_length: Duration):
+        self.forward_market_enabled = True
+        self.slot_length = slot_length
+        self.orders: Dict[
+            str, Dict[int, Dict[DateTime, Dict]]] = defaultdict(lambda: defaultdict(dict))
+        self.previous_device_stats = {}
+        self.current_device_stats: Dict[
+            int, Dict[DateTime, Dict[str, Dict]]] = defaultdict(lambda: defaultdict(dict))
+        self.device_time_series: Dict[
+            int, Dict[int, Dict[DateTime, Dict[str, Dict]]]] = defaultdict(
+            lambda: defaultdict(lambda: defaultdict(dict)))
+        self._total_memory_utilization_kb = 0.0
+
+    # pylint: disable=unused-argument
+    def update(self, area_dict: Dict, core_stats: Dict, current_market_slot: str) -> None:
+        """
+        Update the forward market aggregated results with bids_offers_trades of current slot.
+        """
+        self._clear_device_statistics()
+        for area_uuid, area_result in core_stats.items():
+            if "forward_market_stats" not in area_result:
+                return
+
+            current_market_dt = from_format(current_market, DATE_TIME_FORMAT) if (
+                current_market := current_market_slot) else ""
```

Comment:
I am not very familiar with the `:=` operator and do not really get what is done in this line here. Can you please help me understand? Thanks!

Reply:
Wow, I think that it can be the remnant of previous implementations, anyway it does not make sense to use walrus operator here. Thanks for catching this!

Reply:
I started a new discussion about these lines https://github.com/gridsingularity/gsy-framework/pull/397#discussion_r995747813
marking this as resolved though.